app/gui2/src/util/reactivity.ts

```diff
@@ -150,3 +150,9 @@ export function debouncedGetter<T>(
   })
   return valueRef
 }
+
+/** Update `target` to have the same entries as `newState`. */
+export function syncSet<T>(target: Set<T>, newState: Set<T>) {
+  for (const oldKey of target) if (!newState.has(oldKey)) target.delete(oldKey)
+  for (const newKey of target) if (!target.has(newKey)) target.add(newKey)
+}
```

Comment:
Add unit test for it. It may sound strange, but I actually had an issue in one of this kind of methods discovered long after implementation.

Reply:
Good call, caught a bug actually